crates/openvino/src/tensor.rs

```diff
@@ -48,44 +48,39 @@ impl Tensor {
         let buffer = data.as_ptr() as *mut std::os::raw::c_void;
         try_unsafe!(ov_tensor_create_from_host_ptr(
             element_type,
-            shape.instance(),
+            shape.as_c_struct(),
             buffer,
             std::ptr::addr_of_mut!(tensor)
         ))?;
-        Ok(Self { instance: tensor })
+        Ok(Self { ptr: tensor })
     }
 
     /// Get the pointer to the underlying OpenVINO tensor.
-    pub(crate) fn instance(&self) -> *mut ov_tensor_t {
-        self.instance
+    pub(crate) fn as_ptr(&self) -> *mut ov_tensor_t {
```

Comment:
I noticed several `as_ptr()` and `as_c_struct()`, would it be better to remove the `as_` for getting the underlying values?

Reply:
I'm following the Rust convention for naming pointer-returning functions (e.g., [`slice::as_ptr`](https://doc.rust-lang.org/std/primitive.slice.html#method.as_ptr)). I guess technically I should use `as_mut_ptr` since this returns a `*mut` pointer.

Reply:
The `as_c_struct` is just for naming consistency but also to highlight that these wrapper `struct`s don't contain pointers but rather entire C structs.